TabletDriverFilters/TabletDriverNoiseReduction.cs

```diff
@@ -0,0 +1,189 @@
+using System;
+using System.Collections.Generic;
+using TabletDriverPlugin.Attributes;
+using TabletDriverPlugin.Tablet;
+
+namespace TabletDriverPlugin
+{
+    [PluginName("TabletDriver Noise Reduction")]
+    public class TabletDriverNoiseReduction : IFilter
+    {
+        private LinkedList<Point> _buffer = new LinkedList<Point>();
```

Comment:
Any reason for using a LinkedList? I'm unfamiliar with the type.

Reply:
It's a performance optimization, fast removal of first element.

Reply:
Better performance compared to something like `Queue<T>` or `Stack<T>`? Or is it used in different ways that those can't?

Reply:
Wait, I didn't knew about `Queue<T>`, but it should have the same performance, should I reimplement it as `Queue<T>`?

Reply:
If it suits the needs, its a lot cleaner to call `Enqueue()`, `Dequeue()`, or `Peek()` instead of manually rotating since it handles it internally.

Reply:
Tried reimplementing it into Queue but came up as almost the exact same code, with the only difference being Dequeue() returning a point that the code does not use by design. Will keep it as LinkedList. Unfortunately NoiseReduction can't use Queue's characteristics yet.